### PR TITLE
Atom47 overhaul, add Rev4 and VIA

### PR DESCRIPTION
**Pull request by evyd13**
## Description

I've given the Atom47 a bit of an overhaul by adding the split space and full space layouts. The changes have been applied in all revisions and I've compiled all combinations of revisions and keymaps, which work without errors. I have also added a VIA keymap as well as Rev4. (both have been tested!)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/maartenwut/atom47/rev4/rev4.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#ifndef ATOM47_REV4_H
+#define ATOM47_REV4_H
+
```
Comment (fauxpark) [inline keyboards/maartenwut/atom47/rev4/rev4.h:3]:
```suggestion
#pragma once
```

File: keyboards/maartenwut/atom47/rev4/rev4.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  k30, k31, k32,                k37,           k39, k3a, k3b, k3c \
+) \
+{ \
+  {k00, k01, k02, k03, k04, k05, k06, k07, k08, k09, k0a, k0b, k0c}, \
+  {k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, XXX, k1c}, \
+  {k20, XXX, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b, k2c}, \
+  {k30, k31, k32, XXX, XXX, XXX, XXX, k37, XXX, k39, k3a, k3b, k3c}  \
+}
+
+#endif
```
Comment (fauxpark) [inline keyboards/maartenwut/atom47/rev4/rev4.h:37]:
```suggestion
```

File: keyboards/maartenwut/atom47/rev4/rules.mk [hunk 3]
```diff
@@ -0,0 +1,4 @@
+MCU = atmega32u2
+ENCODER_ENABLE = yes
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality (+1150)
+RGBLIGHT_ENABLE = no   # Enable keyboard underlight functionality
```
Comment (fauxpark) [inline keyboards/maartenwut/atom47/rev4/rules.mk:4]:
```suggestion
# MCU name
MCU = atmega32u2

# Build Options
ENCODER_ENABLE = yes
BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no   # Enable keyboard underlight functionality
```

File: keyboards/maartenwut/atom47/rev4/rev4.h (hunk 1 above)
Comment (Maartenwut) [reply keyboards/maartenwut/atom47/rev4/rev4.h:3]: Why not for the other revisions?

Comment (fauxpark) [reply keyboards/maartenwut/atom47/rev4/rev4.h:3]: I can't add suggestions on unchanged files/hunks.

Comment (Maartenwut) [reply keyboards/maartenwut/atom47/rev4/rev4.h:3]: So I should do it on the other revisions as well?

Comment (fauxpark) [reply keyboards/maartenwut/atom47/rev4/rev4.h:3]: That would probably be a good idea, yeah.

**Comment by tzarc** [comment]
This has a bunch of conflicts -- probably due to the name change.

Is it still relevant? If so, can we get copyright headers added to the sources files that don't include them?

**Comment by evyd13** [comment]
> 
> 
> This has a bunch of conflicts -- probably due to the name change.
> 
> Is it still relevant? If so, can we get copyright headers added to the sources files that don't include them?

No, it's superseded by https://github.com/qmk/qmk_firmware/pull/13201

**Pull request closed without merging**